coffee-chats/pom.xml

```diff
@@ -51,8 +51,111 @@
           <deploy.version>1</deploy.version>
+          <sourceDirectory>${project.build.directory}/${project.build.finalName}/build</sourceDirectory>
         </configuration>
       </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-war-plugin</artifactId>
+        <version>3.3.1</version>
+        <configuration>
+          <warSourceDirectory>${basedir}/src/main/webapp/build</warSourceDirectory>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <groupId>com.github.eirslett</groupId>
+        <artifactId>frontend-maven-plugin</artifactId>
+        <!-- Use the latest released version:
+        https://repo1.maven.org/maven2/com/github/eirslett/frontend-maven-plugin/ -->
+        <version>1.10.0</version>
+
+        <executions>
+          <!-- Installing node and npm -->
+          <execution>
+              <id>install node and npm</id>
+              <goals>
+                  <goal>install-node-and-npm</goal>
+              </goals>
+              <phase>generate-resources</phase>
+
+              <configuration>
+                <nodeVersion>v12.18.3</nodeVersion>
+              </configuration>
+          </execution>
+
+          <!-- Running npm install -->
+          <execution>
+            <id>npm install</id>
+            <goals>
+                <goal>npm</goal>
+            </goals>
+            <phase>generate-resources</phase>
+            
+            <configuration>
+                <arguments>install</arguments>
+            </configuration>
+          </execution>
+
+          <!-- Running frontend tests -->
+          <execution>
+            <id>npm test</id>
+            <goals>
+                <goal>npm</goal>
+            </goals>
+            <phase>test</phase>
+
+            <configuration>
+              <arguments>test</arguments>
+              <environmentVariables>
+                  <CI>true</CI>
+              </environmentVariables>
+            </configuration>
+          </execution>
+
+          <execution>
+            <id>npm build</id>
+            <goals>
+                <goal>npm</goal>
+            </goals>
+            <phase>compile</phase>
+            <configuration>
+                <arguments>run build</arguments>
+            </configuration>
+          </execution>
+        </executions>
+
+        <configuration>
+          <workingDirectory>src/main/webapp</workingDirectory>
+          <installDirectory>target</installDirectory>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+        <version>1.2.1</version>
+        <executions>
+          <execution>
+            <id>appengine-web config</id>
+            <phase>compile</phase>
+            <goals>
+              <goal>exec</goal>
+            </goals>
+
+            <configuration>
+              <executable>cp</executable>
```

Comment:
I would suggest using http://maven.apache.org/plugins/maven-resources-plugin/index.html for copying files, but feel free to ignore since `cp` is available everywhere we want to run maven